packages/components/src/system-message/system-message.tsx

```diff
@@ -0,0 +1,160 @@
+import { AddUserIcon, DeleteIcon, PinIcon } from '@status-im/icons/20'
+import { styled } from '@tamagui/core'
+import { View } from 'react-native'
+
+import { IconAvatar } from '../avatar'
+import { AddedUsersMessageContent } from './components/added-users-message-content'
+import { DeletedMessageContent } from './components/deleted-message-content'
+import { PinnedMessageContent } from './components/pinned-message-content'
+
+type User = {
+  id: string
+  name: string
+  src: string
+}
+
+type SystemMessageType = 'pinned' | 'deleted' | 'added'
+type SystemMessageState = 'default' | 'pressed' | 'landed'
+
+type Props = {
+  state?: SystemMessageState
+  timestamp: string
+} & (
+  | {
+      type: 'pinned'
+      actor: User
+      message: {
+        author: User
+        text: string
+        images?: {
+          type: 'photo' | 'gif'
+          srcs: Array<string>
+        }
+      }
+    }
+  | {
+      type: 'deleted'
+      text: string
+      action?: {
+        label: string
+        onClick: () => void
+      }
+    }
+  | {
+      type: 'added'
+      actor: User
+      users: Array<User>
+    }
+)
+
+const getIcon = (type: SystemMessageType, state: SystemMessageState) => {
```

Comment:
Maybe this should live inside each component so we wouldn't need two switch statements?

Reply:
Yeah, for sure. Thanks for noticing. Addressed in [`2420a6e` (#367)](https://github.com/status-im/status-web/pull/367/commits/2420a6e4a69ec74f8fd3c82dc60c6ce397418d87)